File: frontend/src/app/core/models/report.ts
export interface YearlyReportUserInfo {
  accountId: string;
  name: string;
  team: string | null;
}

export interface YearlyReportRow {
  user: YearlyReportUserInfo;
  hoursByMonth: string[];
  yearTotal: string;
  breakdownByActivity: Record<string, string[]>;
}

export interface YearlyReport {
  year: number;
  expectedWorkingDays: number[];
  yearTargetHours: string;
  rows: YearlyReportRow[];
  columnTotals: string[];
  grandTotal: string;
}

export interface WorkingDaysResponse {
  year: number;
  months: number[];
}

File: backend/app/schemas/report.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class YearlyReportUserInfo(BaseModel):
    account_id: str
    name: str
    team: str | None


class YearlyReportRow(BaseModel):
    user: YearlyReportUserInfo
    hours_by_month: list[Decimal]
    year_total: Decimal
    breakdown_by_activity: dict[str, list[Decimal]] = {}


class YearlyReport(BaseModel):
    model_config = ConfigDict()

    year: int
    expected_working_days: list[int]
    year_target_hours: Decimal
    rows: list[YearlyReportRow]
    column_totals: list[Decimal]
    grand_total: Decimal
